plugin-java/src/main/java/appland/execution/AppMapJavaPackageConfig.java

```diff
@@ -220,7 +224,9 @@ private static VirtualFile[] collectSourceRootsWithDependencies(@NotNull Module
                 .flatMap(m -> Arrays.stream(ModuleRootManager.getInstance(m).getContentRoots()))
                 .collect(Collectors.toUnmodifiableSet());
 
-        return Arrays.stream(ProjectRootManager.getInstance(module.getProject()).getContentSourceRoots())
+        return ProjectRootManager.getInstance(module.getProject())
```

Comment:
The changes for this PR are small, but complex.
I like the change to find the best matching AppMap content root. 

After that directory is found, this method `collectSourceRootsWithDependencies` is called with the generic module again. I think that's not sufficient. 
For example, if you executed `tests` inside `waltz-jobs` of the  `waltz` project, the selected "best AppMap content root" would be `/source/waltz`, but here the code is limiting the packages search for module `waltz-jobs` and its dependencies. This would exclude code of the other modules, for example `waltz-common`.

Instead, using the selected "Best AppMap content root" to filter the available source roots of the project should (hopefully) provide the suitable source roots. 

Because I'm not good at explaining this, I've made https://github.com/getappmap/appmap-intellij-plugin/pull/702 to show some code. Please feel free to use it or to make your own changes where necessary.

(There's a possible edge-case for modules, which have several source roots and which have source roots under "Best AppMap content root" and also source roots outside the "Best AppMap content root". But I'd like to get your opinion on the notes above before we're discussing rather weird edge-cases).